test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts

```diff
@@ -0,0 +1,297 @@
+						expect( tracksEventProperties.post_type ).toBe( 'post' );
+					} );
+
+					it( '"blog_id" is a numeric value', async function () {
+						expect( numericRegex.test( tracksEventProperties.blog_id ) ).toBe( true );
+					} );
+
+					it( '"site_type" matches the current site type', async function () {
+						const expectedSiteType: SiteType = envVariables.TEST_ON_ATOMIC ? 'atomic' : 'simple';
+						expect( tracksEventProperties.site_type ).toBe( expectedSiteType );
+					} );
+
+					it( '"user_locale" is a valid locale', async function () {
+						expect( expectedLocales.has( tracksEventProperties.user_locale ) ).toBe( true );
+					} );
+
+					it( '"blog_tz" is a numeric value', async function () {
+						expect( numericRegex.test( tracksEventProperties.blog_tz ) ).toBe( true );
+					} );
+
+					it( '"user_lang" is a valid locale', async function () {
+						expect( expectedLocales.has( tracksEventProperties.user_lang ) ).toBe( true );
+					} );
+
+					it( '"blog_lang" is a valid locale', async function () {
+						expect( expectedLocales.has( tracksEventProperties.blog_lang ) ).toBe( true );
+					} );
+				} );
+			} );
+		} );
+
+		describe( 'In the page editor', function () {
+			const accountName = getTestAccountByFeature( features );
+			let page: Page;
+			let editorPage: EditorPage;
+			let editorTracksEventManager: EditorTracksEventManager;
+
+			beforeAll( async () => {
+				page = await browser.newPage();
+
+				const testAccount = new TestAccount( accountName );
+				await testAccount.authenticate( page );
+
+				editorTracksEventManager = new EditorTracksEventManager( page );
+				editorPage = new EditorPage( page, { target: features.siteType } );
+			} );
+
+			it( 'Start a new page', async function () {
+				await editorPage.visit( 'page' );
+				await editorPage.waitUntilLoaded();
+				// We'll be leaving without saving here too.
+				editorPage.allowLeavingWithoutSaving();
+			} );
+
+			it( 'Clear the current Tracks events for a clean slate', async function () {
+				await editorTracksEventManager.clearEvents();
+			} );
+
+			describe( 'From adding a page template', function () {
+				it( 'Add "Two column about me layout" page template', async function () {
+					const editorFrame = await editorPage.getEditorHandle();
+					const pageTemplateModalComponent = new PageTemplateModalComponent( editorFrame, page );
+					await pageTemplateModalComponent.selectTemplateCatagory( 'About' );
+					await pageTemplateModalComponent.selectTemplate( 'Two column about me layout' );
+				} );
+
+				it( '"wpcom_block_inserted" event fires with "from_template_selector" set to true', async function () {
+					const eventDidFire = await editorTracksEventManager.didEventFire(
+						'wpcom_block_inserted',
+						{
+							matchingProperties: { from_template_selector: true },
+						}
+					);
+					expect( eventDidFire ).toBe( true );
+				} );
+			} );
+		} );
+
+		describe( 'In the site editor', function () {
```

Comment:
The rename throws off the diff here a bit. Other than some re-organization, here below is the only really meaningful addition to this script.

Reply:
I assume we talked about this at some point, but this is a long test file. Have you found this to be reliable?

Reply:
Yep, so far! The length is slightly misleading as it's really three distinct test flows in one file. Any one flow isn't too long and they've been reliable.

I've been trying to strike a balance between keeping the files lean, but also not making too many (e.g. I think that one file per event might grow a bit too much). How do you think that balance has been working so far? I'm very open to breaking this out into smaller files!

Let me know what you think @worldomonation and I can do that in a future pr! 😄